### PR TITLE
TF: purge `TFTrainer`

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

Removes `TFTrainer` and all its traces.

### CODE REVIEW

File: docs/source/it/migration.md [hunk 1]
```diff
@@ -1,320 +1,313 @@
-<!---
```
Comment (gante) [inline docs/source/it/migration.md:1]: IDK what happened with this diff: the only change is the removal of L169-174 in the old file

File: examples/legacy/text-classification/run_tf_text_classification.py [hunk 2]
```diff
@@ -1,313 +0,0 @@
-#!/usr/bin/env python
```
Comment (gante) [inline examples/legacy/text-classification/run_tf_text_classification.py:1]: This file uses TFTrainer

File: examples/legacy/token-classification/run_tf_ner.py [hunk 3]
```diff
@@ -1,310 +0,0 @@
-#!/usr/bin/env python
```
Comment (gante) [inline examples/legacy/token-classification/run_tf_ner.py:1]: This file uses TFTrainer

**Review by Rocketknight1 — APPROVED** [review]
100% approval

**Review by amyeroberts — APPROVED** [review]
All LGTM - thanks for cleaning all of this up! 🧹 🧹 🧹

**Pull request merged**